docs/rules/group-exports.md

```diff
@@ -0,0 +1,62 @@
+# group-exports
+
+Reports when multiple named exports or CommonJS assignments are present in a single file and when the default export is not adjacent to the named export.
+
+**Rationale:** An `export` declaration or `module.exports` assignment can appear anywhere in the code. By requiring a single export declaration along with the default export being placed next to the named export, your exports will remain at one place, making it easier to see what exports a module provides.
+
+## Rule Details
+
+This rule warns whenever a single file contains multiple named exports or assignments to `module.exports` (or `exports`) and when the default export is not adjacent to the named export.
+
+### Valid
+
+```js
+// Default export statement is adjacent to named export -> ok
+export default function test() {}
+// A single named export statement -> ok
+export const valid = true
+```
+
+```js
+const first = true
+const second = true
+
+// A single named export statement -> ok
+export {
+  first,
+  second,
+}
+```
+
+```js
+// A single exports assignment -> ok
+module.exports = {
+  first: true,
+  second: true
+}
+```
+
+### Invalid
+
+```js
+// Multiple named export statements -> not ok!
+export const first = true
+export const second = true
+```
+
+```js
+// Default export is not adjacent to the named export statement -> not ok!
+export default {}
+const first = true
+export { first }
+```
+
+```js
+// Multiple exports assignments -> not ok!
+exports.first = true
+exports.second = true
```

Comment:
This is a it more complicated for CommonJS. If you wish for your module to export a function, but also some other things under "named" exports

```js
module.exports = function foo() {};
module.exports.first = true;
module.exports.second = true;
```

You could potentially do this like in the following example, but that's pretty odd to satisfy this rule.
```js
module.exports = Object.assign(function foo {}, {first: true, second: true});
```

IMO, this rule should either:
- only reports errors for ES modules
- only report the use of several instructions like `module.exports.foo = true` if there is no other direct assignment (`module.exports = foo`). <-- I'd go for this as I myself prefer grouping `module.exports = { foo, bar }` when possible.

Reply:
Has this been addressed?

Reply:
Yes - the rule now does not care what is being assigned to `module.exports`. Some tests for that behaviour can be [found here](https://github.com/benmosher/eslint-plugin-import/pull/721/files#diff-e516480e5c5622695cf87cc6d15e177fR176-212).